drivers/can/can_mcp2515.c

```diff
@@ -417,7 +418,11 @@ static int mcp2515_send(struct device *dev, const struct zcan_frame *msg,
 	/* Address Pointer selection */
 	abc = 2 * tx_idx;
 
-	mcp2515_cmd_load_tx_buffer(dev, abc, tx_frame, sizeof(tx_frame));
+	/* Calculate minimum length to transfer */
+	len = sizeof(tx_frame) - MCP2515_DLC_MAX;
+	len += MIN(tx_frame[MCP2515_FRAME_OFFSET_DLC] & 0x0F, MCP2515_DLC_MAX);
```

Comment:
Do you really need the `MIN(..)` here?
What do you think about
```C
len += msg->dlc;
```

Reply:
I agree with @alexanderwachter, an assert check here should be enough. 

The frame converter code hardcaps the memory copy at 8 both ways, now also nicer with your changes :-) In itself that shouldn't corrupt anything beyond D7. Maybe limit the DLC/len field there as well, the converters are used both in rx and tx direction. Nice catch with loopback, I thought about the situation and assumed the bus would 'clear it up'

Reply:
@karstenkoenig as far as I can tell there's no opportunity for MCP2515 register corruption (from a user's application) via CAN RX but via TX there is.  At present the frame converter doesn't hardcap the DLC field at 8, it applies a 4-bit mask:
https://github.com/zephyrproject-rtos/zephyr/blob/acef1574beb0cbcf940be1f40ad74aea8df142a0/drivers/can/can_mcp2515.c#L215
If the TX zcan_frame DLC field dictates the number of data bytes the MCP2515 registers are at risk of corruption from the user's application.  An error in the user's application (or a bit flip) could create a DLC in the range 0x09 to 0x0F.

The MCP2515 will not bork if the DLC is greater than 8 but likely will if unknown stack data is copied beyond the D7 register.

I'm not sure if there would be a reason to use a DLC > 8, possibly it could be useful for fuzzing other CAN interfaces?  I'm not sure.  Thus the proposal for the LOG_WRN instead of assert.  The MCP2515 datasheet makes a note that the DLC can be set to > 8 but doesn't say it shouldn't be done.

Possibly a better and safer solution (and a louder failure that something is wrong for the application developer) is to detect the invalid DLC field at the mcp2515_send function entry, regard the request potentially corrupt and return CAN_ERR or possibly -EINVAL.  Considering a general Zephyr goal is to be [safety certified](https://www.zephyrproject.org/zephyr-project-rtos-first-functional-safety-certification-submission-for-an-open-source-real-time-operating-system/) this would be reasonable.

Reply:
@nixward Can you add a check for DLC > 8 in send(...) ?

Reply:
@nixward Sorry I think I wasn't clear. I agree with you any value of DLC over 8 is bogus and should be dropped. It's not a use case we need to be concerned with, we don't have to unlock the full (and still pretty limited) capabilities of the chip where there is nothing to be gained :-)
I was merely suggesting to just hard cap the DLC values in the converter methods like this, which will cover both RX and TX. 
https://github.com/karstenkoenig/zephyr/commit/9ef4059e22061ce6e6b1881e15dfcdba25177566

Any further error returning or logging I wouldn't necessarily object to, but I am concerned that this wouldn't be consistent across the board. e.g. the stm32 driver is already implementing a check but is only treating it as an assert.

Reply:
@karstenkoenig no problem, I've updated the commit with @alexanderwachter 's #20367 PR as a dependency.  So as long as that PR goes in first the corruption issue will be resolved.